go/cmd/dolt/commands/dump.go

```diff
@@ -467,3 +476,36 @@ func addBulkLoadingParadigms(dEnv *env.DoltEnv, fPath string) error {
 
 	return writer.Close()
 }
+
+// addCreateDatabaseHeader adds a CREATE DATABASE header to prevent `no database selected` errors on dump file ingestion.
+func addCreateDatabaseHeader(dEnv *env.DoltEnv, fPath, dbName string) error {
+	writer, err := dEnv.FS.OpenForWriteAppend(fPath, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	_, err = writer.Write([]byte("CREATE DATABASE IF NOT EXISTS " + dbName + "; USE " + dbName + "; \n"))
```

Comment:
(minor) It might be worth wrapping `dbName` in backtick/identifier quotes. That way we can support databases with MySQL reserved words that are allowed to be used, but must be backtick/identifier quoted. For example, `interval` is a reserved MySQL word that needs to be backtick quoted. Would be good to add a test for that case, too. 

MySQL docs on keywords and reserved words: https://dev.mysql.com/doc/refman/8.0/en/keywords.html